src/folders.ts

```diff
@@ -82,7 +82,7 @@ export class CMakeToolsFolderController implements vscode.Disposable {
    * @param ws The workspace folder to search
    */
   get(ws: vscode.WorkspaceFolder | undefined): CMakeToolsFolder | undefined {
-    return ws ? this._instances.get(ws.uri.fsPath) : ws;
+    return (ws && ws.uri) ? this._instances.get(ws.uri.fsPath) : undefined;
```

Comment:
I believe the path is passed in as element 2 in an array for the "run command from task" case.  This function should accept a new type for `ws` which is an array of two elements.

Reply:
Thank you for suggestion - it's totally right, I've updated PR... And I guess we need test suite for tasks now.